osu.Game/Overlays/Dashboard/CurrentlyOnlineDisplay.cs

```diff
@@ -104,6 +113,36 @@ protected override void OnFocus(FocusEvent e)
             searchTextBox.TakeFocus();
         }
 
+        private void onBlocksUpdated(object? sender, NotifyCollectionChangedEventArgs e)
+        {
+            switch (e.Action)
+            {
+                case NotifyCollectionChangedAction.Add:
+                    Debug.Assert(e.NewItems != null);
+
+                    foreach (APIRelation block in e.NewItems.Cast<APIRelation>())
+                    {
+                        int userId = block.TargetID;
+                        removeUserPanel(userId);
```

Comment:
Is this something that should even be handled client-side at all like this? I'd say no, and based on https://github.com/ppy/osu-server-spectator/issues/264 I'd say I'm not alone in thinking that.